tests/listobject_test.py

```diff
@@ -1,32 +1,205 @@
-from tests.utility import *
 from unittest import *
 import botocore
 import botocore.session
-
+from libs.execute import *
+from libs.command import *
+from libs.utility import *
+from unittest import *
+from datetime import datetime
 
 class ListObject_Test(TestCase):
+
+    # ======== Construction, setUp, tearDown =========
+
+    def __init__(self, *args, **kwargs):
+        self.bucket_irods_path = '/tempZone/home/alice'
+        self.bucket_name = 'alice-bucket'
+        self.key = 's3_key2'
+        self.secret_key = 's3_secret_key2'
+        self.s3_bridge_url = 'http://localhost:8080'
+        super(ListObject_Test, self).__init__(*args, **kwargs)
+
     def setUp(self):
-        mkdir("")
-        mkdir("a-first-test")
-        mkdir("b-second-test")
 
-        for i in range(10):
-            touch_file(f"a-first-test/{i}")
-            touch_file(f"b-second-test/{i}")
+        # create collections/data objects
+        make_local_file('f1', 100)
+        make_local_file('f2', 200)
+
+        assert_command('imkdir %s/dir1' % self.bucket_irods_path)
+        assert_command('iput f1 %s/f1' % self.bucket_irods_path)
+        assert_command('iput f1 %s/dir1/d1f1' % self.bucket_irods_path)
+        assert_command('iput f2 %s/dir1/d1f2' % self.bucket_irods_path)
+        assert_command('imkdir %s/dir1/dir1a' % self.bucket_irods_path)
+        assert_command('iput f1 %s/dir1/dir1a/d1af1' % self.bucket_irods_path)
+        assert_command('iput f2 %s/dir1/dir1a/d1af2' % self.bucket_irods_path)
+        assert_command('imkdir %s/dir1/dir1b' % self.bucket_irods_path)
+        assert_command('iput f1 %s/dir1/dir1b/d1bf1' % self.bucket_irods_path)
+        assert_command('iput f2 %s/dir1/dir1b/d1bf2' % self.bucket_irods_path)
+        assert_command('imkdir %s/dir2' % self.bucket_irods_path)
 
         session = botocore.session.get_session()
         self.client = session.create_client("s3",
                                             use_ssl=False,
-                                            endpoint_url="http://127.0.0.1:8080",  # normal networking stuff :p
-                                            aws_access_key_id="no",
-                                            aws_secret_access_key="heck")
-        set_access("", 'own',recursive=True)
-        mkdir("", access_level="own")
+                                            endpoint_url=self.s3_bridge_url,  # normal networking stuff :p
+                                            aws_access_key_id=self.key,
+                                            aws_secret_access_key=self.secret_key)
```

Comment:
If there are things in `setUp` or `tearDown` that never change, consider moving those things into `setUpClass` and `tearDownClass`.

That'll make tests complete sooner.

Reply:
Yes, that will help.  Will do.

Reply:
Done